api/queries_pr.go

```diff
@@ -394,12 +394,17 @@ func PullRequestForBranch(client *Client, repo ghrepo.Interface, branch string)
 	}
 
 	for _, pr := range resp.Repository.PullRequests.Nodes {
-		if pr.HeadLabel() == branch {
+		if pr.HeadLabel() == headBranch {
+			if baseBranch != "" {
+				if pr.BaseRefName != baseBranch {
```

Comment:
Nit: couldn't have all this be collapsed into a single conditional?

Reply:
they could; i often over-nest because i find it easier to read and reason about